addons/payment_nuvei/tests/test_payment_transaction.py

```diff
@@ -0,0 +1,146 @@
+from unittest.mock import patch
+
+from werkzeug import urls
+
+from odoo.tests import tagged
+from odoo.tools import mute_logger
+
+from odoo.addons.payment import utils as payment_utils
+
+from odoo.addons.payment_nuvei.controllers.main import NuveiController
+from odoo.addons.payment_nuvei.tests.common import NuveiCommon
+
+
+@tagged('post_install', '-at_install')
+class TestPaymentTransaction(NuveiCommon):
+
+    def _get_expected_values(self, tx):
+        return_url = self._build_url(NuveiController._return_url)
+        first_name, last_name = payment_utils.split_partner_name(tx.partner_name)
+
+        webhook_url = self._build_url(NuveiController._webhook_url)
+        cancel_url_params = {
+            'tx_ref': tx.reference,
+            'return_access_tkn': self._generate_test_access_token(
+                self.provider.nuvei_secret_key,
+                tx.reference,
+            ),
+        }
+
+        phone_number = tx.partner_phone and tx._phone_format(
+            number=tx.partner_phone, country=tx.partner_country_id, raise_exception=False
+        )
+
+        return {
+            'address1': tx.partner_address,
+            'city': tx.partner_city,
+            'country': tx.partner_country_id.code,
+            'currency': tx.currency_id.name,
+            'email': tx.partner_email,
+            'encoding': 'UTF-8',
+            'first_name': first_name,
+            'item_amount_1': tx.amount,
+            'item_name_1': tx.reference,
+            'item_quantity_1': 1,
+            'invoice_id': tx.reference,
+            'last_name': last_name,
+            'merchantLocale': tx.partner_lang,
+            'merchant_id': self.provider.nuvei_merchant_identifier,
+            'merchant_site_id': self.provider.nuvei_site_identifier,
+            'payment_method_mode': 'filter',
+            'payment_method': 'unknown',
+            'phone1': phone_number,
+            'state': tx.partner_state_id.code or '',
+            'time_stamp': tx.create_date.strftime('%Y-%m-%d.%H:%M:%S'),
+            'total_amount': self.amount,
+            'version': '4.0.0',
+            'zip': tx.partner_zip,
+            'notify_url': webhook_url,
+            'success_url': return_url,
+            'error_url': f'{return_url}?{urls.url_encode(cancel_url_params)}',
+            'pending_url': return_url,
+            'back_url': f'{return_url}?{urls.url_encode(cancel_url_params)}',
+        }
+
+    def test_no_item_missing_from_rendering_values(self):
+        """ Test that the rendered values are conform to the transaction fields. """
+
+        tx = self._create_transaction(flow='redirect')
+        with patch(
+            'odoo.addons.payment.utils.generate_access_token', new=self._generate_test_access_token
+        ):
+            processing_values = tx._get_specific_rendering_values(None)
+        expected_values = self._get_expected_values(tx)
+        self.assertDictEqual(processing_values['url_params'], expected_values)
```

Comment:
What about the `api_url` and `checksum`? You can compare all the rendering values, not just the `url_params` ^^